### ISSUE TITLE
Blacklisting tags via tagNamePreference causes false errors

### ISSUE THREAD

**Issue by jaydenseric** [gajus/eslint-plugin-jsdoc#332]
With:

```json
{
    "settings": {
        "jsdoc": {
            "tagNamePreference": {
                "description": false
            }
        }
    }
}
```

The `jsdoc/require-description` rule reports an `` Unexpected tag `@description` `` error, highlighting the whole JSDoc comment block, for every block regardless if `@description` is present.

With:

```json
{
    "settings": {
        "jsdoc": {
            "tagNamePreference": {
                "implements": false
            }
        }
    }
}
```

The same thing happens, but with the `jsdoc/implements-on-classes` rule and an `` Unexpected tag `@implements` `` error.

<img width="589" alt="Screen Shot 2019-07-12 at 6 33 23 pm" src="https://user-images.githubusercontent.com/1754873/61114406-99637e80-a4d3-11e9-806d-4ff8018c1dcf.png">

I am attempting to ban both of those tags and more via `tagNamePreference`, to match the [`jsdoc-md` list of supported tags](https://github.com/jaydenseric/jsdoc-md#tag-subset).

Those two tags are the only ones that seem to cause rules to break.
